### ISSUE TITLE
Communitas map crashing when launching a new game

### ISSUE THREAD

**Issue by Magean0** [LoneGazebo/Community-Patch-DLL#2609]
_1. Mod version (i.e Date - 4/23):_

30/09

_2. Mod list (if using Vox Populi only, leave blank):_

InfoAddict, Quick Turns, RED Modpack, RED - just ethnic diversity

_3. Error description:_ 

When I click start to launch a new game on the Communitas map script, the game seems to be loading, but before I get the leader-specific loading screen, the game crashes. I tried on the previous 19/09 version, and everything works fine. Reinstalling the last version again causes the crash to occur once more. 

_4. Steps to reproduce (optional):_

Specifically, I had selected Carthage, on a Large map, 11 Civs and 22 city-states, with Low Sea level, and Terra placement (aka everyone on the same continent). Everything left as default.

[logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/517291/logs.zip)

**Comment by LoneGazebo** [comment]
You are getting some database errors from RED.

**Comment by thunderbird14** [comment]
Ethnic Diversity and RED modpacks conflict each other.

**Comment by Magean0** [comment]
Thanks! So, that's it. But, how comes a graphical unit pack prevents one specific map from working, and not all of them? I was able to start a game on Planet Simulator, for instance. Besides, it worked fine with Communitas on the previous version. I'll go reinstall the RED mods to check.

**Comment by Heinous-Hat** [comment]
I think thunderbird14 is thinking of the standalone  Ethnic Diversity unit texture mod. What Magean0 is referencing is the mod-mod for R.E.D. (by nutty) which re-scales units and requires R.E.D. to operate. I use it as well with no issues, although of late I've run into memory issues and leaderhead corruption when using any unit texture mod on big maps with high #of players.  Also, I think R.E.D. has always thrown some innocuous db errors.

I'll try starting a game with Communitas and those mods and post back, but I think this is a... R.E.D. herring ;-)

Forgive me

**Comment by Magean0** [comment]
Right, I'm referencing the mod-mod for RED rescaling units, not the graphical pack competing with RED called Ethnic Diversity (the "mod-mod" has a similar name as it removes the downsizing effect of RED while keeping only the diversity thing).

However, it still seems to be the source of the crash. I ran a series of test: first, I used only VP's six components, then I added Quick Turns and InfoAddict, next I added RED. So far, so good. At that point, I added "RED just ethnic diversity" and got a CTD. The strange thing is that it works with other maps, and worked on previous versions.

**Comment by Heinous-Hat** [comment]
Ok, seeing the same thing now (10/6 beta). R.E.D. and J.E.D. (v.27) together will produce CTD when generating a Communitas map (only). Works alright with R.E.D. by itself... likewise if you just deactivate the VP collection. 

There are 6 invalid ArtDefine references produced by the R.E.D. J.E.D. combination. J.E.D. causes this, but to my knowledge this has always been the case. May be an indicator in this scenario... or maybe benign. I'll ask nutty about these.

In my experience of late, using **any** unit texture mods with VP will cause graphical corruption at some point in the game. Personally, I get leaderhead scene errors and/or crashes during lh scenes. Could be we're seeing a common memory problem and this particular combination is pushing it to the surface earlier.

**Comment by Magean0** [comment]
In my case, no, I'm not using any terrain mod.

Are you suggesting that, even without CTDs, texture mods can lead to memory leaks or a general slowdown of the game?

**Comment by Heinous-Hat** [comment]
Yes, there's a longstanding correlation between unit texture mods and corruption of UI graphics (i.e. wrong textures). At least, that's what I can reproduce on my own machine under certain conditions. 

No performance hit. Just texture problems and related CTDs (usually descriptive).

**Comment by LoneGazebo** [comment]
Sorry, I can't debug other mods. If it is crashing, don't use the crashing mod.

**Comment by Heinous-Hat** [comment]
It definitely predates Community Patch, but it seems to be manifesting in different ways with VP active.

**Comment by Vyyt02** [comment]
Probably related - I was getting constant CTDs while generating huge Communitas maps with certain settings before. I could not narrow down the settings causing it, but there were certain setups working fine and then I did something random, like reducing the number of city states and I got a CTD - always. It was quite some time ago (I am mostly using the Tectonic map these days), but I am sure I had the Ethnic Units mod active at that time.

As for the memory leaks / graphic glitches, I am sometimes getting these on huge maps with the Ethnic Units in later game stages too. When it becomes too bad and too frequent, I simply deactivate the Ethnic Units and I can carry on after a reload.
